bin/pre-commit-hook.js

```diff
@@ -20,45 +28,51 @@ console.log(
 // Make quick pass over config files on every change
 require( '../server/config/validate-config-keys' );
 
+const dirtyFiles = new Set(
+	execSync( 'git diff --name-only --diff-filter=ACM' )
+		.toString()
+		.split( '\n' )
+		.map( name => name.trim() )
+		.filter( name => name.endsWith( '.js' ) || name.endsWith( '.jsx' ) )
+		.map( file => path.join( __dirname, '../', file ) )
+);
+
 const files = execSync( 'git diff --cached --name-only --diff-filter=ACM' )
 	.toString()
 	.split( '\n' )
 	.map( name => name.trim() )
 	.filter( name => name.endsWith( '.js' ) || name.endsWith( '.jsx' ) );
 
-/**
- * Returns true if the given text contain `@format`
- * within its first docblock. False otherwise.
- *
- * @param {String} text text to scan for the format keyword within the first docblock
- */
-const shouldFormat = text => {
-	const firstDocBlockStartIndex = text.indexOf( '/**' );
-
-	if ( -1 === firstDocBlockStartIndex ) {
-		return false;
-	}
-
-	const firstDocBlockEndIndex = text.indexOf( '*/', firstDocBlockStartIndex + 1 );
-
-	if ( -1 === firstDocBlockEndIndex ) {
-		return false;
-	}
-
-	const firstDocBlockText = text.substring( firstDocBlockStartIndex, firstDocBlockEndIndex + 1 );
-	return firstDocBlockText.indexOf( '@format' ) >= 0;
-};
-
 // run prettier for any files in the commit that have @format within their first docblock
 files
-	.map( file => path.join( __dirname, '../', file ) )
-	.forEach( file => {
-		const text = fs.readFileSync( file, 'utf8' );
+	.map( file => [ file, path.join( __dirname, '../', file ) ] )
+	.forEach( ( [ file, fullPath ] ) => {
+		if ( dirtyFiles.has( fullPath ) ) {
```

Comment:
Why do you need to compare the full paths here? Even the relative paths (i.e., direct output of `git diff`) should be comparable to each other, shouldn't they?

Reply:
In 84fe0dd I made a change to parse `git diff` in one consistent way, always to absolute paths.

You're right that it shouldn't matter whether path comparison is relative or absolute as long as they're the same.

I've gone with absolute paths everywhere because that's what was used for the file reading/writing. Some absolute path lookup could be avoided, but I suspect that has little benefit.